packages/nextjs/components/scaffold-stark/Balance.tsx

```diff
@@ -16,21 +18,45 @@ type BalanceProps = {
  * Display (ETH & USD) balance of an ETH address.
  */
 export const Balance = ({ address, className = "", usdMode }: BalanceProps) => {
+  const price = useGlobalState((state) => state.nativeCurrencyPrice);
   const { targetNetwork } = useTargetNetwork();
+  const [balance, setBalance] = useState("0");
+  const [displayUsdMode, setDisplayUsdMode] = useState(
+    price > 0 ? Boolean(usdMode) : false,
+  );
 
-  const price = useGlobalState((state) => state.nativeCurrencyPrice);
-  const {
-    data: balance,
-    isError,
-    isLoading,
-  } = useBalance({
-    address,
+  const { data, isLoading, isError } = useContractRead({
+    address:
+      "0x49D36570D4E46F48E99674BD3FCC84644DDD6B96F7C741B1562B82F9E004DC7",
```

Comment:
move this out to some `constants.ts` cannot be hardcoded in the hook. maybe we can have `utils/eth.ts` and have both `ethAbi` and `ethAddr`